gluten-core/src/main/scala/io/glutenproject/execution/SortExecTransformer.scala

```diff
@@ -52,123 +51,36 @@ case class SortExecTransformer(
 
   override def output: Seq[Attribute] = child.output
 
-  override def outputOrdering: Seq[SortOrder] = sortOrder
+  override def outputOrdering: Seq[SortOrder] = child match {
+    case project: ProjectExecTransformer if ProjectTypeHint.isPreProject(project) =>
+      getSortOrderFromPreProject(project.projectList)
+    case InputIteratorTransformer(InputAdapter(r2c: RowToColumnarExecBase)) =>
+      r2c.child match {
+        case project: ProjectExec if ProjectTypeHint.isPreProject(project) =>
+          // This case means pre-project is fallback.
+          getSortOrderFromPreProject(project.projectList)
+        case _ => sortOrder
+      }
+    case _ => sortOrder
+  }
```

Comment:
I am trying to understand the ordering replacement logic here.

I noticed Spark has a `sameExpression` argument in `SortOrder`:

https://github.com/apache/spark/blob/e428fe902bb1f12cea973de7fe4b885ae69fd6ca/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala#L66

Would that possibly provide a way to simplify this part of code?

Reply:
`sameOrderExpressions` will change the children of SortOrder, it is not very consistent with the original `sortOrder` we need to restore.

Reply:
My previous understanding was mistaken; it appears that sameOrderExpressions can be used to compare changes in sortOrder.

Reply:
> it is not very consistent with the original sortOrder we need to restore

Any reason we have to exactly use the original sort order? I assume the ordering information should be enriched after adding pre-project. E.g. `Sort(a, b, c=(a+b))` could be considered sorted by `a+b` or `c` at the same time.

Reply:
> My previous understanding was mistaken; it appears that sameOrderExpressions can be used to compare changes in sortOrder.

I see. And please ignore my previous comment.